uber/site_sections/merch_admin.py

```diff
@@ -0,0 +1,302 @@
+                        merch_items.insert(0, 
+                                           'WARNING: Attendee is not fully paid up and may not have paid for their merch. '
+                                           'Please contact Registration.')
+
+        return {
+            'id': id,
+            'shirt': shirt,
+            'message': message,
+            'merch_items': merch_items,
+            'gets_swadge': gets_swadge,
+            'swadges_available': c.SWADGES_AVAILABLE
+        }
+
+    @ajax
+    def give_merch(self, session, id, shirt_size, no_shirt, staff_merch, give_swadge=None):
+        try:
+            shirt_size = int(shirt_size)
+        except Exception:
+            shirt_size = None
+
+        success = False
+        attendee = session.attendee(id, allow_invalid=True)
+        merch = attendee.staff_merch if staff_merch else attendee.merch
+        got = attendee.got_staff_merch if staff_merch else attendee.got_merch
+        if not merch:
+            message = '{} has no merch'.format(attendee.full_name)
+        elif got and give_swadge and not attendee.got_swadge:
+            message = '{a.full_name} marked as receiving their swadge'.format(
+                a=attendee)
+            success = True
+            attendee.got_swadge = True
+            session.commit()
+        elif got:
+            message = '{} already got {}'.format(attendee.full_name, merch)
+        elif shirt_size == c.SIZE_UNKNOWN:
+            message = 'You must select a shirt size'
+        else:
+            if no_shirt:
+                message = '{} is now marked as having received all of the following (EXCEPT FOR THE SHIRT): {}'
+            else:
+                message = '{} is now marked as having received {}'
+            message = message.format(attendee.full_name, merch)
+            setattr(attendee,
+                    'got_staff_merch' if staff_merch else 'got_merch', True)
+            if give_swadge:
+                attendee.got_swadge = True
+            if shirt_size:
+                attendee.shirt = shirt_size
+            if no_shirt:
+                session.add(NoShirt(attendee=attendee))
+            success = True
+            session.commit()
+
+        return {
+            'id': id,
+            'success': success,
+            'message': message
+        }
+
+    @ajax
+    def take_back_merch(self, session, id, staff_merch=None):
+        attendee = session.attendee(id, allow_invalid=True)
+        if staff_merch:
+            attendee.got_staff_merch = False
+        else:
+            attendee.got_merch = attendee.got_swadge = False
+            c._swadges_available = False  # force db check next time
+        if attendee.no_shirt:
+            session.delete(attendee.no_shirt)
+        session.commit()
+        return '{a.full_name} ({a.badge}) merch handout canceled'.format(a=attendee)
+
+    @ajax
+    def redeem_merch_discount(self, session, badge_num, apply=''):
```

Comment:
FWIW, I'm pretty sure that as of last year we no longer use this page, so we should probably hide it.  No need to do that as part of this PR though.